ledger/src/blockstore.rs

```diff
@@ -1132,19 +1102,23 @@ impl Blockstore {
             e.num_repaired += 1;
         }
 
-        // Should be safe to modify index_meta here. Two cases
-        // 1) Recovery happens: Then all inserted erasure metas are removed
-        // from just_received_coding_shreds, and nothing will be committed by
-        // `check_insert_coding_shred`, so the coding index meta will not be
-        // committed
+        // insert coding shred into rocks
+        metrics.num_inserted += 1;
         index_meta.coding_mut().set_present(shred_index, true);
+        let result = self
+            .insert_coding_shred(index_meta, &shred, write_batch)
```

Comment:
`insert_coding_shred` is already doing `set_present`:
https://github.com/solana-labs/solana/blob/a272e19f8/ledger/src/blockstore.rs#L1291
so no need to do `set_present` here.

Also, `set_present` should actually be done only if write is successful.